synapse/rest/admin/users.py

```diff
@@ -683,3 +684,29 @@ async def on_PUT(self, request, user_id):
         await self.store.set_server_admin(target_user, set_admin_to)
 
         return 200, {}
+
+
+class UserMembershipRestServlet(RestServlet):
+    """
+    Get room list of an user.
+    """
+
+    PATTERNS = admin_patterns("/users/(?P<user_id>[^/]+)/rooms$")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.store = hs.get_datastore()
+
+    async def on_GET(self, request, user_id):
+        await assert_requester_is_admin(self.auth, request)
+
+        if not self.hs.is_mine(UserID.from_string(user_id)):
```

Comment:
We have a preference for trying to avoid storing `self.hs`, as it can lead to circular dependencies not caught during start up due to the dependency injections. This is technically fine to do so here, and I know we do it elsewhere, but could we change this to specifically storing the function rather than `hs`? i.e.:

```python
    def __init__(self, hs):
        self.is_mine = hs.is_min
        ...

    async def on_GET(self, request, user_id):
        ....
        if not self.is_mine(UserID.from_string(user_id)):
            ...
```